crates/libs/core/src/sync/bridge.rs

```diff
@@ -0,0 +1,61 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+use std::future::Future;
+
+use mssf_com::FabricCommon::{
+    IFabricAsyncOperationCallback, IFabricAsyncOperationContext, IFabricAsyncOperationContext_Impl,
+};
+
+use crate::runtime::executor::Executor;
+use windows_core::AsImpl;
+
+use super::bridge_context::BridgeContext;
+
+/// Bridge for SF Begin com api.
+/// This launches the future task on rt, and triggers the callback when
+/// the future completes.
+/// The returned context must be passed up to the Begin api's return.
+/// The End api must use fabric_end_bridge() to retrieve the async result from
+/// the returned context.
+/// Remarks:
+/// This api in a sense is unsafe because the develper needs to ensure the following:
```

Comment:
What are the consequences of violating this invariant?

Reply:
Access violation or memory corruption. I may add some debug checks in future.